### ISSUE TITLE
ACCURACY

### ISSUE THREAD

**Issue by deepdarklowtech** [cmu-l3/l1#19]
I noticed that in your paper, you use accuracy as the evaluation metric, but in the code, it seems to be using pass@16. Could you clarify whether this is equivalent to accuracy?

Thank you!

**Comment by Pranjal2041** [comment]
Hi, in code we report both Pass@1 and Pass@16. In the paper, we report Pass@1, which is equivalent to accuracy.

**Comment by deepdarklowtech** [comment]
However, it appears that the code calculates pass@1 by averaging the scores instead of applying greedy decoding, which may not align with the standard interpretation of accuracy.

**Comment by deepdarklowtech** [comment]
In fact, when evaluating the Qwen-1.5B model on the Math500 dataset, I was not able to replicate the results presented in your paper.

**Comment by Pranjal2041** [comment]
> averaging Pass@1:

Averaging 16 Pass@1 scores is equivalent to computing accuracy across 16 different seeds, followed by averaging to reduce variance. Greedy decoding is the independent concept of accuracy and is typically not used for evaluating reasoning models. We have used temperature=0.6 for both training and inference in line with previous works.

>  evaluating the Qwen-1.5B model on the Math500 dataset

Can you please provide the details of the evaluation, what scripts you used, and what scores you got?

**Comment by deepdarklowtech** [comment]
accuracy= 42%， average tokens = 458

[before_math500_qwen1.5b.json](https://github.com/user-attachments/files/20087505/before_math500_qwen1.5b.json)
```
def reranking_length(completions):
    def last_boxed_only_string(string):
        idx = string.rfind("\\boxed")
        if "\\boxed " in string:
            return "\\boxed " + string.split("\\boxed ")[-1].split("$")[0]
        if idx < 0:
            idx = string.rfind("\\fbox")
            if idx < 0:
                return None

        i = idx
        right_brace_idx = None
        num_left_braces_open = 0
        while i < len(string):
            if string[i] == "{":
                num_left_braces_open += 1
            if string[i] == "}":
                num_left_braces_open -= 1
                if num_left_braces_open == 0:
                    right_brace_idx = i
                    break
            i += 1
    
        if right_brace_idx is None:
            retval = None
        else:
            retval = string[idx:right_brace_idx + 1]
    
        return retval
    
    def remove_boxed(s):
        if s is None:
            return None
        if "\\boxed " in s:
            left = "\\boxed "
            assert s[:len(left)] == left
            return s[len(left):]
    
        left = "\\boxed{"
    
        assert s[:len(left)] == left
        assert s[-1] == "}"
    
        return s[len(left):-1]
    
    def extract_solution(solution_str: str) -> str:
        """Extract the final boxed solution from a solution string.
    
        Args:
            solution_str: Raw solution string that may contain multiple boxed answers
    
        Returns:
            The final boxed answer with box notation removed
        """
        return remove_boxed(last_boxed_only_string(solution_str))
    
    return extract_solution(completions)

test =  load_dataset("HuggingFaceH4/MATH-500")['test']
gen_config = GenerationConfig(
    do_sample=False,       
    num_beams=1,
    max_new_tokens=4096,
    bos_token_id=151646,   
    eos_token_id=151643
    
)

for tmp in test:
    print(f"[{i+1}/{total_sample}] processing...", flush=True)
    i+=1
    question = tmp['problem']+"Let's think step by step and output the final answer within \\boxed{}. "
    inputs = tokenizer(question, return_tensors="pt").to("cuda:0")
    l1 = len(inputs[0])
    with torch.no_grad():
        out = model.generate(
                    **inputs,
                    pad_token_id=tokenizer.pad_token_id,
                    generation_config=gen_config
                )
    length = len(out[0])-l1
    generated = tokenizer.batch_decode(out, skip_special_tokens=True)[0]
    answer = str(tmp['answer'])
    try:

        if reranking_length(generated) == answer:
            a+=1
            d = dict()
            d['question'] = question
            d['length'] = length
            d['solution'] = generated[len(question):]
            ret.append(d)
    except Exception as e:
        print(e)

with open("math500_qwen1.5b.json", "w") as f:
    json.dump(ret, f, indent=4)

```

**Comment by Pranjal2041** [comment]
Can you please share the model used? This looks like Qwen-1.5B and unrelated to L1. You can check their technical report for their reported results. You can also serve it using VLLM and then use the `run_llama.py` script in the repository. 

Since it is unrelated to L1, I am closing the issue now! For L,1 you can use the replication scripts provided in `scripts/replicate` directory. If the issue persists there, please feel free to reopen.
